tests/resources/dynamic_config.py

```diff
@@ -0,0 +1,33 @@
+import pandas as pd
+
+from hamilton.function_modifiers import ResolveAt, dynamic, extract_columns, parameterize, source
+
+
+@extract_columns("a", "b", "c", "d", "e")
+def df() -> pd.DataFrame:
+    """Produces a dataframe with columns a, b, c, d, e consisting entirely of 1s"""
+    return pd.DataFrame(
+        {
+            "a": [1],
+            "b": [1],
+            "c": [1],
+            "d": [1],
+            "e": [1],
+        }
+    )
+
+
+@dynamic(
+    lambda columns_to_sum_map: parameterize(
+        **{
+            key: {"col_1": source(value[0]), "col_2": source(value[1])}
+            for key, value in columns_to_sum_map.items()
+        }
+    ),
+    at=ResolveAt.COMPILE,
+)
```

Comment:
if someone has the flag off -- will Hamilton parse this without breaking?

My assumption is that, with the flag off -- Hamilton will just make this a no-op and drop it from the DAG? or?

Reply:
Currently it'll break when the driver is instantiated, but it works at import time

Reply:
also, what happens when there is no `columns_to_sum_map` in the config passed in? Should it fail? or should the node not resolve? My thinking is that it should not resolve to anything.

Reply:
I think it should be fine to have it, with it off if the flag is off. If anyone depends on it -- they'll get an error saying that it's not satisfied. If they don't need it to run, then it wont error.

Reply:
I think that it should fail -- otherwise its strange DAG behavior. And these are likely to be central to it -- if one inherits it, it should work, otherwise it'll be confusing.